index.js

```diff
@@ -20,31 +20,38 @@ module.exports = function(options) {
     options.watch(client);
   }
 
-  return through.obj(function transform(file, enc, callback) {
-    if (file.isStream()) {
-      return callback(new PluginError('gulp-scp2', 'Streaming not supported.'));
-    }
-
-    var path = join(options.dest, file.relative);
-    client.mkdir(dirname(path), function(err) {
-      if (err) {
-        return callback(new PluginError('gulp-scp2', err));
+  return through.obj(function(file, enc, callback) {
+      if (file.isStream()) {
+        return callback(new PluginError('gulp-scp2', 'Streaming not supported.'));
       }
 
-      client.write({
-        destination: path,
-        content: file.contents
-      }, function(err) {
+      var filePath = pathLib.relative(process.cwd(), file.path);
```

Comment:
Why not use `file.relative`?  the path of the file is not determined by where you run the command.